test/functional/legacy/065_float_and_logic_operators_spec.lua

```diff
@@ -0,0 +1,182 @@
+-- Test for floating point and logical operators.
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('floating point and logical operators', function()
+  setup(clear)
+
+  it('is working', function()
+    insert([=[
+      Results of test65:]=])
+
+    source([[
+      $put =printf('%f', 123.456)
+      $put =printf('%e', 123.456)
+      $put =printf('%g', 123.456)]])
+    -- Check we don't crash on division by zero.
+    source([[
+      echo 1.0 / 0.0
+      $put ='+='
+      let v = 1.234
+      let v += 6.543
+      $put =printf('%g', v)
+      let v = 1.234
+      let v += 5
+      $put =printf('%g', v)
+      let a = 5
+      let a += 3.333
+      $put =string(a)
+      $put ='=='
+      let v = 1.234
+      $put =v == 1.234
+      $put =v == 1.2341
+      $put ='add-subtract'
+      $put =printf('%g', 4 + 1.234)
+      $put =printf('%g', 1.234 - 8)
+      $put ='mult-div'
+      $put =printf('%g', 4 * 1.234)
+      $put =printf('%g', 4.0 / 1234)
+      $put ='dict'
+      $put =string({'x': 1.234, 'y': -2.0e20})
+      $put ='list'
+      $put =string([-123.4, 2.0e-20])
+      $put ='abs'
+      $put =printf('%d', abs(1456))
+      $put =printf('%d', abs(-4))
+    ]])
+    -- The test seems to fail if this execute call is put inside the above or
+    -- the below source() call.
+    execute([[$put =printf('%d', abs([1, 2, 3]))]])
```

Comment:
It seems to me that the test fails if I put this line of vim code from a "execute" into a "source" call. I also tested it with this short file:
```lua
local helpers = require('test.functional.helpers')
local clear, source = helpers.clear, helpers.source
local execute, expect = helpers.execute, helpers.expect
describe('text on stderr', function()
  setup(clear)
  it('is working', function()
    execute('put =printf("%d", [1])')
  end)
  it('is not working', function()
    source('put =printf("%d", [1])')
  end)
end)
```
If I save this to a file (say `test/functional/legacy/source_error_spec.lua`)
and then execute `make
TEST_FILE=test/functional/legacy/source_error_spec.lua`, make will run the
first test and display a green dot.  Then it will hang (I assume it is doing
the second test)

This is actually different than what I see with the test65.  If I put the line
in question into the source call the makefile reports:
```
[SNIP]
✱
0 successes / 0 failures / 1 error / 0 pending : 0.001139 seconds

Error → ...functional/legacy/065_float_and_logic_operators_spec.lua @ 10
floating point and logical operators is working
./test/functional/helpers.lua:67: Vim(put):E745: Using a List as a Number

stack traceback:
        ./test/functional/helpers.lua:67: in function 'request'
        ./test/functional/helpers.lua:125: in function 'nvim_command'
        ./test/functional/helpers.lua:215: in function 'source'
        ...functional/legacy/065_float_and_logic_operators_spec.lua:14: in func
tion <...functional/legacy/065_float_and_logic_operators_spec.lua:10>

-- Output to stderr:

CMake Error at /home/luc/vcs/neovim/cmake/RunTests.cmake:39 (message):
  Running functional tests failed with error: 1.


CMakeFiles/functionaltest.dir/build.make:51: recipe for target 'CMakeFiles/func
tionaltest' failed
[SNIP]
```